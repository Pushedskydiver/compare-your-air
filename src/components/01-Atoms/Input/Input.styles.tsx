import styled from 'styled-components';
import { colors } from '../../../styles/colors.styles';
import { typography } from '../../../styles/typography.styles';
import spacing from '../../../styles/spacing.styles';

const InputStyles = styled.input`
  position: relative;
  width: 100%;
  padding-top: ${spacing.padding}px;
  padding-right: ${spacing.padding / 2}px;
  padding-bottom: ${spacing.padding}px;
  padding-left: ${spacing.padding * 4}px;
  border: 2px solid ${colors.neutral};
  border-radius: 8px;
  color: ${colors.dark};
  font-size: ${typography.body.size.regular};
  appearance: none;
  z-index: 2;
`;

export default InputStyles;
