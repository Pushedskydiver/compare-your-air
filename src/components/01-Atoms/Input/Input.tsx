import React, { memo, useContext, useState } from 'react';
import InputStyles from './Input.styles';
import Context from '../../Context';

const Input = () => {
  const { selectLocation, searchResults } = useContext(Context);
  const [value, setValue] = useState('Enter city name…');

  function handleChangeEvent(event: any) {
    const value: string = event.target.value;
    const result: any = searchResults.find((item: any) => item.name === value);
    const isMatch = result ? result.name.toLowerCase() : null;

    if (isMatch && isMatch === value.toLowerCase()) {
      selectLocation(value);
    }

    setValue(value);
  }

  function handleBlurEvent() {
    if (value === '') {
      setValue('Enter city name…');
    }
  }

  function handleFocusEvent() {
    if (value === 'Enter city name…') {
      setValue('');
      return;
    }
  }

  return (
    <InputStyles
      data-testid="location-input"
      type="text"
      id="city"
      list="results"
      value={value}
      onChange={handleChangeEvent}
      onBlur={handleBlurEvent}
      onFocus={handleFocusEvent}
    />
  );
}

export default memo(Input);
