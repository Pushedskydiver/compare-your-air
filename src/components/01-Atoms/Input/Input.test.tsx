import React from 'react';
import { cleanup, render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Input from './Input';

afterEach(() => {
  cleanup();
});

describe('<Input />', () => {
  it('should match snapshot', () => {
    const { container } = render(<Input />);

    expect(container).toMatchSnapshot();
  });

  it('should contain value of "Manchester"', async () => {
    const { getByTestId } = render(<Input />);
    const input = getByTestId('location-input');

    await userEvent.type(input, 'Manchester');
    expect(input).toHaveAttribute('value', 'Manchester');
  })
})