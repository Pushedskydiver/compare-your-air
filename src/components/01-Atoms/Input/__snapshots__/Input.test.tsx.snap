// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> should match snapshot 1`] = `
<div>
  <input
    class="sc-AxjAm honbvB"
    data-testid="location-input"
    id="city"
    list="results"
    type="text"
    value="Enter city nameâ€¦"
  />
</div>
`;
